win32ss/user/ntuser/message.c

```diff
@@ -484,31 +484,30 @@ CopyMsgToKernelMem(MSG *KernelModeMsg, MSG *UserModeMsg, PMSGMEMORY MsgMemoryEnt
             TRACE("Copy Message %u from usermode buffer\n", KernelModeMsg->message);
             /* Don't do extra testing for 1 word messages. For examples see
              * https://wiki.winehq.org/List_Of_Windows_Messages. */
-            if (Size > 1)
+            if ((Size > 1) && UserModeMsg->lParam)
             {
                 WCHAR lParamMsg[_countof(StrUserKernel[0]) + 1] = { 0 };
                 _SEH2_TRY
                 {
-                    if (UserModeMsg->lParam)
-                        RtlCopyMemory(lParamMsg, (WCHAR*)UserModeMsg->lParam, sizeof(lParamMsg));
-                        /* Make sure that the last WCHAR is a UNICODE_NULL */
-                        for (i = 0; i < ARRAYSIZE(lParamMsg); ++i)
-                        {
-                            if (lParamMsg[i] == 0)
-                                break;
-                        }
-                        /* If we did not find a UNICODE_NULL, then set last WCHAR to one */
-                        if (i == ARRAYSIZE(lParamMsg))
-                            lParamMsg[_countof(StrUserKernel[0])] = UNICODE_NULL;
+                    RtlCopyMemory(lParamMsg, (WCHAR*)UserModeMsg->lParam, sizeof(lParamMsg));
                 }
                 _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
                 {
                     _SEH2_YIELD(return STATUS_ACCESS_VIOLATION);
                 }
                 _SEH2_END;
 
-                if (UserModeMsg->lParam && !UserModeMsg->wParam &&
-                    PosInArray(lParamMsg) >= 0)
+                /* Make sure that the last WCHAR is a UNICODE_NULL */
+                for (i = 0; i < ARRAYSIZE(lParamMsg); ++i)
+                {
+                    if (lParamMsg[i] == 0)
+                        break;
+                }
+                /* If we did not find a UNICODE_NULL, then set last WCHAR to one */
+                if (i == ARRAYSIZE(lParamMsg))
+                    lParamMsg[_countof(StrUserKernel[0])] = UNICODE_NULL;
```

Comment:
Why `_countof(StrUserKernel[0])`? `ARRAYSIZE(lParamMsg)` seems better to me.

Reply:
a question to @SergeGautherie : do you have any specific reason to favor ARRAYSIZE over _countof?

ARRAYSIZE has to be defined somewhere and requires a few more keys to stroke than _countof. That are 2 drawbacks of ARRAYSIZE. I am not sure though, whether it might have any intrinsic benefits over _countof().
`ARRAYSIZE(lParamMsg) - 1`
`_countof(lParamMsg) - 1`
Do both result in the same binary size when - let's say - applied 1000 times within a binary?

Reply:
My point is not about which function/macro to use.
It is about being consistent on these 2 lines and not/less depend on the variable size definition.

Reply:
I have now removed the "if (i...", so there is no longer a disagreement here. Thanks.

Reply:
Now I am using ```lParamMsg``` and ```ARRAYSIZE(lParamMsg) - 1```. Thanks.